Modules/Utils/Sources/PocketCastsUtils/Extensions/Equatable+When.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+public extension Equatable {
+    /// `.when` allows you to perform the `action` closure when the object and `value` are equal.
+    ///
+    /// While this essentially a wrapper around an is equal check this helps reduce some boilerplate when you want to
+    /// both return the value and perform an action if that value is equal to something.
+    ///
+    /// Examples:
+    ///
+    ///     // Before:
+    ///     func doSomething() -> Bool {
+    ///         let success = database.addTheThing()
+    ///
+    ///         if success { sendSomething() }
+    ///
+    ///         return success
+    ///     }
+    ///
+    ///     // After:
+    ///     func doSomething() -> Bool {
+    ///         database.addTheThing().when(true) {
```

Comment:
[nit] While reading it for the first time my brain went: "when true? true what?"

I wonder if we could just do a `database.addTheThing().succeeds { }` or `succeeded { }`, this way it's pretty clear that the block will be executed when `addTheThing()` succeeds and there's no margin to interpret from where the `true` will come from.

Reply:
I initially went the direction of adding a private extension for Bool that adds an `onSuccess` but I feel like I can see wanting to solve this again in the future outside of just Bool, especially as we move towards more async functional type programming. 

This is why I went with the `.when` extension on Equatable. However I do see your point about the lack of context when first looking at the code. 

Would renaming it from `when` to something more event based like `on` would make more sense? To me seeing `.on(true) { ... }` solves that. 

Ideally in the future it wouldn't be as "generic" as `.on(true)` and would be something more like `something.on(.success){ ... }`.